sktree/ensemble/_honest_forest.py

```diff
@@ -0,0 +1,295 @@
+
+    classes_ : ndarray of shape (n_classes,) or a list of such arrays
+        The classes labels (single output problem), or a list of arrays of
+        class labels (multi-output problem).
+
+    n_classes_ : int or list
+        The number of classes (single output problem), or a list containing the
+        number of classes for each output (multi-output problem).
+
+    n_features_ : int
+        The number of features when ``fit`` is performed.
+
+        .. deprecated:: 1.0
+            Attribute `n_features_` was deprecated in version 1.0 and will be
+            removed in 1.2. Use `n_features_in_` instead.
+
+    n_features_in_ : int
+        Number of features seen during :term:`fit`.
+
+    feature_names_in_ : ndarray of shape (`n_features_in_`,)
+        Names of features seen during :term:`fit`. Defined only when `X`
+        has feature names that are all strings.
+
+        .. versionadded:: 1.0
+
+    n_outputs_ : int
+        The number of outputs when ``fit`` is performed.
+
+    feature_importances_ : ndarray of shape (n_features,)
+        The impurity-based feature importances.
+        The higher, the more important the feature.
+        The importance of a feature is computed as the (normalized)
+        total reduction of the criterion brought by that feature.  It is also
+        known as the Gini importance.
+
+        Warning: impurity-based feature importances can be misleading for
+        high cardinality features (many unique values). See
+        :func:`sklearn.inspection.permutation_importance` as an alternative.
+
+    oob_score_ : float
+        Score of the training dataset obtained using an out-of-bag estimate.
+        This attribute exists only when ``oob_score`` is True.
+
+    oob_decision_function_ : ndarray of shape (n_samples, n_classes) or \
+            (n_samples, n_classes, n_outputs)
+        Decision function computed with out-of-bag estimate on the training
+        set. If n_estimators is small it might be possible that a data point
+        was never left out during the bootstrap. In this case,
+        `oob_decision_function_` might contain NaN. This attribute exists
+        only when ``oob_score`` is True.
+
+    honest_decision_function_ : ndarray of shape (n_samples, n_classes) or \
+            (n_samples, n_classes, n_outputs)
+        Decision function computed on each sample, including only the trees
+        for which it was in the honest subsample. It is possible that a sample
+        is never in the honest subset in which case `honest_decision_function_`
+        might contain NaN. This attribute exists when `honest_score_` is True.
+
+    structure_indices_ : list of lists, shape=(n_estimators, n_structure)
+        Indices of training samples used to learn the structure
+
+    honest_indices_ : list of lists, shape=(n_estimators, n_honest)
+        Indices of training samples used to learn leaf estimates
+
+    Notes
+    -----
```

Comment:
We can mention in the docstring `Notes` something along these lines:

```
Honesty is a feature of trees that enables unbiased estimates of confidence intervals(?) <some description>
with citation to Athey's paper(s). The default implementation here is using double sampling to implement honesty. The amount of samples used for learning split nodes vs leaf nodes is controlled by the ``honest_fraction`` parameter.

This forest classifier is a "meta-estimator" because any tree model can be used in the classification process,
while enabling honesty separates the data used for split and leaf nodes.
```

Reply:
Or some paraphrased version of that.